index/index.go

```diff
@@ -0,0 +1,49 @@
+package index
+
+import (
+	"fmt"
+	"os"
+	"io/ioutil"
+	"text/template"
+
+	"pkg.re/essentialkaos/ek.v9/fsutil"
+
+	"github.com/gongled/vgrepo/storage"
+)
+
+// ////////////////////////////////////////////////////////////////////////////////// //
+
+func ExportIndex(index *storage.VStorage, templateFile string, outputFile string) error {
+	if templateFile == "" {
+		return fmt.Errorf("Can't use given template")
+	}
+
+	if fsutil.IsExist(outputFile) {
+		err := os.Remove(outputFile)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	fd, err := os.OpenFile(outputFile, os.O_CREATE|os.O_WRONLY, 0644)
+
+	if err != nil {
+		return err
+	}
+
+	defer fd.Close()
+
+	tpl, err := ioutil.ReadFile(templateFile)
```

Comment:
You should move reading template and checking error before `OpenFile` part. Because if an error occurs while reading template, `OpenFile` will create an empty output file.